lib/cn/kvset.c

```diff
@@ -522,54 +486,28 @@ kvset_open2(
 
     assert(ks->ks_kvsetid != 0);
 
-    /* map single hblock */
-    err = mpool_mcache_mmap(mp, 1, &km->km_hblk_id, &ks->ks_hmap);
+    err = kvset_hblk_init(mp, km->km_hblk_id, &ks->ks_vgmap, &ks->ks_use_vgmap,
+        &ks->ks_hlog, &ks->ks_hblk);
     if (ev(err))
         goto err_exit;
 
-    {
-        struct mblock_props props;
-
-        err = mpool_mblock_props_get(mp, km->km_hblk_id, &props);
-        if (ev(err))
-            goto err_exit;
-
-        err = kvset_hblk_init(mp, &props, ks->ks_hmap, &ks->ks_vgmap, &ks->ks_use_vgmap,
-                              &ks->ks_hlog, &ks->ks_hblk);
-
-        if (ev(err))
-            goto err_exit;
-
-        /* kvset_stats from kblocks */
-        ks->ks_st.kst_halen += props.mpr_alloc_cap;
-        ks->ks_st.kst_hwlen += props.mpr_write_len;
-        ks->ks_st.kst_ptombs += ks->ks_hblk.kh_metrics.hm_nptombs;
-    }
+    /* kvset_stats from hblocks */
+    ks->ks_st.kst_halen += ks->ks_hblk.kh_hblk_desc.alen_pages * PAGE_SIZE;
+    ks->ks_st.kst_hwlen += ks->ks_hblk.kh_hblk_desc.wlen_pages * PAGE_SIZE;
+    ks->ks_st.kst_ptombs += ks->ks_hblk.kh_metrics.hm_nptombs;
 
     ks->ks_seqno_min = ks->ks_hblk.kh_seqno_min;
     ks->ks_seqno_max = ks->ks_hblk.kh_seqno_max;
     assert(ks->ks_seqno_min <= ks->ks_seqno_max);
 
-    /* map kblocks */
-    if (km->km_kblk_list.idc > 0) {
-        err = mpool_mcache_mmap(mp, km->km_kblk_list.idc, km->km_kblk_list.idv, &ks->ks_kmap);
-        if (ev(err))
-            goto err_exit;
-    }
-
     kcachesz = 0;
 
-    for (i = 0; i < n_kblks; i++) {
+    for (uint i = 0; i < n_kblks; i++) {
```

Comment:
Isn't n_kblks a uint32_t

Reply:
I'd love to have a compiler option to warn if `sizeof(i) < sizeof(n_blks)`?  Manually keeping track of scalar widths is error prone.

Reply:
Made this change.  Your language server must be smarter than mine. :)

Reply:
That is partially what `-Wconversion` is partially for. Though it wouldn't warn on our 64 bit linux systems, it would probably warn on a system where uint != 32 bits.

I just know we have a lot of mismatched integer types and I have touched this area of code, so I know when to warn you!

Reply:
Right, it prob won't warn because technically fixed-width types are an illusion as they are defined in terms of the basic C types (char, short, int, long, ..) with some knowledge of the compiilation environment architecture.  So uint == uint32_t, but uint32_t is a typedef so surely the compiler could tell the difference? But I would imagine -Wconversion would be completely useless on most code in that case...